### DISCUSSION TITLE
How to test exact match with `it.each`

### DISCUSSION THREAD

**Discussion by Barbapapazes** [vitest-dev/vitest#6048] [Q&A]
Hello 👋,

I'm using this test:

```ts
it
	.each([
		['foobar']
	])
	('should match result', (result) => {
		expect(() => willThrowAnError()).toThrowError(result)
	})
```

My issue is that if the function `willThrowAnError` throw an error with the message `foo`, it will pass because `toThrowError` does not check for exact match.

By reading the docs, https://vitest.dev/api/expect.html#tothrowerror, I see I can use a regex but it does not works because I should built it dynamically using `new RegExp` and this will try to match `/foobar/` which is not expected.

How could I have test exact match?

**Comment by RobinTail** [comment]
For instance, @Barbapapazes :

```ts
try {
  willThrowError();
  assert.fail("Must throw error");
} catch (e) {
  expect(e).toBeInstanceOf(ParticularErrorClass); // if you're using custom error classes
  expect(e.message).toBe("particular text");
}
```

> How could I have test exact match?

OR you can use RegExp with `^` and `$` as it's noted in the example you refer to:

```ts
// Test the exact error message
expect(() => getFruitStock('pineapples')).toThrowError(
  /^Pineapples are not in stock$/,
)
```

As per your case it would be:
```ts
it.each([ /^foobar$/ ])
```
